src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -44,37 +39,83 @@ public MovieSession add(MovieSession movieSession) {
     }
 
     @Override
-    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+    public Optional<MovieSession> get(Long id) {
         try (Session session = HibernateUtil.getSessionFactory().openSession()) {
-            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
-            CriteriaQuery<MovieSession> criteriaQuery =
-                    criteriaBuilder.createQuery(MovieSession.class);
-            Root<MovieSession> root = criteriaQuery.from(MovieSession.class);
-            Predicate moviePredicate = criteriaBuilder.equal(root.get("movie").get("id"), movieId);
-            Predicate datePredicate = criteriaBuilder.between(root.get("showTime"),
-                    date.atStartOfDay(), date.atTime(END_OF_DAY));
-            Predicate allConditions = criteriaBuilder.and(moviePredicate, datePredicate);
-            criteriaQuery.select(root).where(allConditions);
-            root.fetch("movie");
-            root.fetch("cinemaHall");
-            return session.createQuery(criteriaQuery).getResultList();
+            return Optional.ofNullable(session.get(MovieSession.class, id));
         } catch (Exception e) {
-            throw new DataProcessingException("Can't get available sessions for movie with id: "
-                    + movieId + " for date: " + date, e);
+            throw new DataProcessingException("Can't get a movieSession by id: " + id, e);
         }
     }
 
     @Override
-    public Optional<MovieSession> get(Long id) {
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
         try (Session session = HibernateUtil.getSessionFactory().openSession()) {
-            Query<MovieSession> query = session.createQuery("FROM MovieSession ms "
-                    + "JOIN FETCH ms.movie m "
-                    + "JOIN FETCH ms.cinemaHall ch "
-                    + "WHERE ms.id = :id", MovieSession.class);
-            query.setParameter("id", id);
-            return query.uniqueResultOptional();
+            LocalDateTime start = date.atTime(0,0,0);
+            LocalDateTime end = date.atTime(23, 59, 59);
+            Query<MovieSession> movieSessionQuery
+                    = session.createQuery("from MovieSession ms "
+                            + "where ms.movie.id = :movieId "
+                            + "and ms.showTime > :start "
+                            + "and ms.showTime < :end",
+                    MovieSession.class);
+            movieSessionQuery.setParameter("movieId", movieId);
+            movieSessionQuery.setParameter("start", start);
+            movieSessionQuery.setParameter("end", end);
+            return movieSessionQuery.getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException(
+                    "Can't get available sessions on the scheduled date: " + date, e);
+        }
+    }
+
+    @Override
+    public boolean update(MovieSession movieSession) {
```

Comment:
Consider changing the return type of the `update` method to `void`. The current approach of returning a boolean by comparing entities is not reliable for determining if the update was successful.